### DISCUSSION TITLE
500 internal server

### DISCUSSION THREAD

**Discussion by AdelEddarai** [fuma-nama/fumadocs#666] [Q&A] [answered]
### To Reproduce

why i can't fetch the blog in production, it works as expected in local but can't fetch in production t gives 500 internal server 
https://netflow.netlify.app/blog

if u click to any blog it gives u 500 internal server
![Screenshot 2024-08-01 174950](https://github.com/user-attachments/assets/42ddfd62-40c6-4ffb-9554-2bacc9fb5c14)

### Current vs. Expected behavior

please checkout my repo : https://github.com/AdelEddarai/netflow/

### Provide environment information

```bash
nextjs 14
"framer-motion": "^11.2.10",
"fumadocs-core": "^12.5.6",
"fumadocs-mdx": "^8.2.34",
"fumadocs-typescript": "^2.0.1",
"fumadocs-ui": "^12.5.6",
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

it works fine locally but in production it throw 500 internal server

**Comment by fuma-nama** [comment]
Please provide the error logs. 

Notice that Fumadocs is not a blog framework, it's likely that you have made mistakes in your code, which shouldn't be a bug report.

**Comment by krystianmolas** [comment]
I've encountered a similar issue in my project. The only error I see in the server console is:

`
 ⨯ [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: 'DYNAMIC_SERVER_USAGE'
}`

**Reply by krystianmolas** [reply]
Deleting 

`export async function generateStaticParams() {
  return languages.flatMap((lang) =>
    getPages(lang)!.map((page) => ({
      slug: page.slugs,
      lang,
    }))
  );
}`

from [lang]\docs\[[...slug]]\page.tsx fixes issue

**Reply by fuma-nama** [reply]
This doesn't seems to be related to the OP's issue, they've I18n enabled on the entire app while blog implementation is copied from the fumadocs site, which still need to investigate.

If you've encountered similar issues, please open another question with relevant code and reproducible example, I cannot find the issue with only your production error messages.

**Comment by fuma-nama** [comment]
From the source code of your repo, I think you forgot to configure [Static Generation](https://fumadocs.vercel.app/docs/ui/internationalization#static-generation) for I18n. 

Also you can test locally by running `next build` and `next start` if it helps.

I cannot run your repo on dev mode without secret environment variables, please provide a minimal reproducible repo.

**Comment by AdelEddarai** [comment, marked as answer]
Thank you so much I have solved the issue by deleting generateStaticParams
function from [slug]/page.tsx
